odl/contrib/datasets/ct/mayo.py

```diff
@@ -126,74 +139,83 @@ def load_projections(folder, indices=None):
-             ((np.max(angles) - np.min(angles)) / (2 * np.pi)))
+    table_dist = datasets[-1].DetectorFocalCenterAxialPosition - \
+                 datasets[0].DetectorFocalCenterAxialPosition
+    num_rot = (angles[-1] - angles[0]) / (2 * np.pi)
+    pitch = table_dist / num_rot
 
-    # Get flying focal spot data
-    offset_axial = np.array([d.SourceAxialPositionShift for d in datasets])
+    # offsets: detector’s focal center -> focal spot
     offset_angular = np.array([d.SourceAngularPositionShift for d in datasets])
     offset_radial = np.array([d.SourceRadialDistanceShift for d in datasets])
+    offset_axial = np.array([d.SourceAxialPositionShift for d in datasets])
 
-    # TODO(adler-j): Implement proper handling of flying focal spot.
-    # Currently we do not fully account for it, merely making some "first
-    # order corrections" to the detector position and radial offset.
-
-    # Update angles with flying focal spot (in plane direction).
-    # This increases the resolution of the reconstructions.
-    angles = angles - offset_angular
-
-    # We correct for the mean offset due to the rotated angles, we need to
-    # shift the detector.
-    offset_detector_by_angles = det_radius * np.mean(offset_angular)
-    minp[0] -= offset_detector_by_angles
-    maxp[0] -= offset_detector_by_angles
-
-    # We currently apply only the mean of the offsets
-    src_radius = src_radius + np.mean(offset_radial)
+    # angles have inverse convention
+    shift_d = np.cos(-offset_angular) * (src_radius + offset_radial) - src_radius
+    shift_t = np.sin(-offset_angular) * (src_radius + offset_radial)
+    shift_r = + offset_axial
 
-    # Partially compensate for a movement of the source by moving the object
-    # instead. We need to rescale by the magnification to get the correct
-    # change in the detector. This approximation is only exactly valid on the
-    # axis of rotation.
-    mean_offset_along_axis_for_ffz = np.mean(offset_axial) * (
-        src_radius / (src_radius + det_radius))
+    shifts = np.transpose(np.vstack([shift_d, shift_t, shift_r]))
 
     # Create partition for detector
-    detector_partition = odl.uniform_partition(minp, maxp, shape)
+    detector_partition = odl.uniform_partition(det_minp, det_maxp, det_shape)
 
     # Convert offset to odl definitions
-    offset_along_axis = (mean_offset_along_axis_for_ffz +
-                         datasets[0].DetectorFocalCenterAxialPosition -
-                         angles[0] / (2 * np.pi) * pitch)
+    offset_along_axis = datasets[0].DetectorFocalCenterAxialPosition - \
+                        angles[0] / (2 * np.pi) * pitch
 
     # Assemble geometry
     angle_partition = odl.nonuniform_partition(angles)
+
+    # Flying focal spot
+    src_shift_func = None
+    if use_ffs:
+        src_shift_func = lambda angle: odl.tomo.flying_focal_spot(
+                        angle, apart=angle_partition, shifts=shifts)
```

Comment:
- I prefer the construct
  ```py
  if use_ffs:
      # something
  else:
      src_shift_func = None
  ```
  Yours looks a bit like C-style 😉 
- Minor nit (PEP8): rather than assigning a lambda, consider using `functools.partial`:
  ```py
   src_shift_func = partial(odl.tomo.flying_focal_spot, apart=angle_partition, shifts=shifts)
  ```
  I find it a bit easier on the eyes.